packages/adapter-d1/src/d1.ts

```diff
@@ -227,3 +232,341 @@ function onError(error: Error): never {
     message,
   })
 }
+
+/**
+ * HTTP API binding.
+ */
+
+type D1HTTPResponseInfo = {
+  code: number // >= 1000
+  message: string
+}
+
+type D1HTTPRawResult = {
+  meta?: Partial<D1Meta>
+  results?: {
+    columns?: string[]
+    rows?: unknown[][]
+  }
+  success?: boolean
+}
+
+function onHTTPError(error: Error): never {
+  debug('HTTP Error: %O', error)
+  const { message } = error
+
+  throw new DriverAdapterError({
+    kind: 'sqlite',
+    extendedCode: 1,
+    message,
+  })
+}
+
+async function performRawQuery(client: KyInstance, options: KyOptions) {
+  try {
+    const response = (await client.post('raw', options).json()) as {
+      errors: D1HTTPResponseInfo[]
+      messages: D1HTTPResponseInfo[]
+      result: D1HTTPRawResult[]
+      success?: true
+    }
+
+    const tag = '[js::performRawQuery]'
+    debug(`${tag} %O`, {
```

Comment:
This can be inlined.

Reply:
No, that would be inconsistent with the rest of the file